src/pages/schedule.js

```diff
@@ -0,0 +1,143 @@
+import React from 'react';
+import { graphql, Link } from 'gatsby';
+import Helmet from 'react-helmet';
+import MarkdownStyles from '../css/markdown-styles';
+import { Container, SmallTitle, MarkdownContainer } from '../utils/components';
+import Layout from '../utils/components/Layout';
+
+const currentDate = new Date();
+
+export default function Schedule({ data }) {
+  const { children } = data.transcripts.nodes.reduce(
```

Comment:
This reduce reads pretty confusing to me. I'm gonna take a stab at a refactor—with the time comparison, flags, and unused accumulator values, I'm finding it hard to follow the logic. Might try and split it into separate "upcoming/current/past" events with a `filter()`, see how that looks. If it's not a clear win I'll open a PR to review separately.

Reply:
Got something I feel very good about, so I pushed it directly. [Commit](https://github.com/reactiflux/reactiflux.com/pull/96/commits/17c1958562d937dc812c085499e6dad47acf11d3). Stripped it down to a couple helper functions, a component, and a pair of filters.

Reply:
Cool, definitely an improvement! 😁

Wonder if it'd be a good idea to refactor these bits slightly:

https://github.com/reactiflux/reactiflux.com/blob/17c1958562d937dc812c085499e6dad47acf11d3/src/pages/schedule.js#L15-L20

https://github.com/reactiflux/reactiflux.com/blob/17c1958562d937dc812c085499e6dad47acf11d3/src/pages/schedule.js#L24-L29

into something like:

```js
const [upcomingEventNodes, pastEventNodes] = partition(
	(node) =>
          currentDate <= new Date(node.childMarkdownRemark.frontmatter.date),
	nodes.map(simplifyNode)
)

const upcomingEvents = groupBy('dateGroup', upcomingEventNodes)
const pastEvents = groupBy('dateGroup', pastEventNodes)
```

🤔

Reply:
Ooo wait partition sounds awesome, I haven't used that. Yeah I like that